pkg/dogstatsd/replay/writer.go

```diff
@@ -164,7 +171,12 @@ func (tc *TrafficCaptureWriter) Capture(d time.Duration) {
 			return
 		}
 		tc.File = fp
-		tc.writer = bufio.NewWriter(tc.File)
+		if compressed {
+			tc.zWriter = zstd.NewWriter(tc.File)
+			tc.writer = bufio.NewWriter(tc.zWriter)
+		} else {
+			tc.writer = bufio.NewWriter(tc.File)
+		}
```

Comment:
Note: This code is very similar to https://github.com/DataDog/datadog-agent/pull/8699/files#diff-6b9b2309fa9a1802f8a39ae6368bd178f256ae77228e19bb6214ad16bd09729fR159-R164 and can be moved out of the `if` block.

Reply:
You're absolutely right, done.